src/OpalCompiler-Core/OpalCompiler.class.st

```diff
@@ -370,11 +370,14 @@ OpalCompiler >> compile [
 	"Policy: compiling is non-faulty by default"
 	self permitFaulty ifNil: [ self permitFaulty: false ].
 
-	ast
-		ifNil: [
-			result := self parse.
-			ast ifNil: [ "some failBlock" ^ result ] ]
-		ifNotNil: [ ast scope ifNil: [ self doSemanticAnalysis ] ].
+	ast ifNil: [
+		result := self parse.
+		ast ifNil: [ "some failBlock" ^ result ] ].
+
+	self callPlugins.
+	(compilationContext isSemanticAnalysisNeeded or: [ ast scope isNil ]) 
+		ifTrue: [ 
+			self doSemanticAnalysis ].
```

Comment:
I don't think this is necessarily true for all plugins.

There are for sure plugins that do code rewriting and need a semantic analysis afterwards.
But there are others that would need to hook after the semantic analysis.

Also, the `isSemanticAnalysisNeeded` boolean is very low level and dangerous API which opens a can of worms.
Now people may produce wrong compilations very easily.

I understand you need to do something special.
But then why not just drive the compilation yourself in the way you want from outside of this class?
All the low level API to do it is already there

Reply:
well, that's why we added the require semantic analysis flag, because some transformations require it and some others not :P
I was just trying to make the string interpolation plugin to load so I can actually start work on it... I have no idea how I would "drive the compilation myself", I will dig into it.